Metrics generation and visualisation in immuadmin client - Closes #121

padurean opened the pull request.

mmeloni commented on cmd/immuadmin/immuadmin.go:
```diff
@@ -0,0 +1,220 @@
+}
+
+func optionsWithAuth() *client.Options {
+	var token string
```
I believe this can be avoid thanks to the new constructor available in client: client.NewImmuClient(options())

mmeloni commented on pkg/server/server.go:
```diff
@@ -123,6 +140,16 @@ func (s *ImmuServer) Start() error {
+	// your RPCs. However, since it is bad practice to have metrics of high
+	// cardinality the latency monitoring metrics are disabled by default. To
+	// enable them the following has to be called during initialization code:
+	grpc_prometheus.EnableHandlingTimeHistogram()
```
IMHO this has to be enable by config flag

mmeloni commented on pkg/server/server.go:
```diff
@@ -114,7 +126,12 @@ func (s *ImmuServer) Start() error {
+		s.Options.MetricsBind(),
+		s.Logger,
+		func() float64 { return float64(s.Store.CountAll()) },
+		func() float64 { return time.Since(startedAt).Hours() },
```
is possible to use an inline time.Now() and avoid to use a var?

padurean replied:
@mmeloni the `startedAt` value needs to be set only once: when server was started. If a call to `time.Now()` would be used there, the duration returned by `time.Since` will always be almost zero (from `time.Since` godoc: It is shorthand for `time.Now().Sub(t)`, so it will basically compare now with now).

mmeloni commented on cmd/immuadmin/statisticscmd/cmd.go:
```diff
@@ -0,0 +1,160 @@
+		Short:   fmt.Sprintf("Show statistics"),
+		Aliases: []string{"s"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			//--> workaround to achieve auth (this command does HTTP requests which do not go through ImmuClient)
```
metric server should not protected by auth. This maybe is useless

padurean commented:
@mmeloni Thanks for the review!

- Where do you think we should put a more comprehensive message regarding visual statistics option?
- I am not sure why/how `scanln`would prevent exposing sensitive data, can you please explain? Also, i do not know how to achieve that since the args are extracted by the `cobra` library. Also, the error you've shown seems unavoidable, unless the user uses single quotes around the password OR if we change the password generation to not include special characters anymore (i.e. so that password contains just alphanumeric characters).

mmeloni commented:
> @mmeloni Thanks for the review!
> 
> * Where do you think we should put a more comprehensive message regarding visual statistics option?
> * I am not sure why/how `scanln`would prevent exposing sensitive data, can you please explain? Also, i do not know how to achieve that since the args are extracted by the `cobra` library. Also, the error you've shown seems unavoidable, unless the user uses single quotes around the password OR if we change the password generation to not include special characters anymore (i.e. so that password contains just alphanumeric characters).

This are only comments, not change requests, but it can be helpful when we will implement new users auth. scanln read a line until it found \n char . Please see vcn in qhich password is prompted https://github.com/vchain-us/vcn/blob/4037b956ac216c4d8d85e962b1b6e79b9f093b9b/pkg/cmd/internal/cli/credentials.go

padurean commented:
@mmeloni i've fixed all your requests. Regarding the graphical statistics view, i've made it the default mode and i've replaced the `-v` option with a `-t` one which will show the text version.
